headers/ErrorLog.hpp

```diff
@@ -37,16 +34,16 @@ class ErrorLog {
 			std::cout << "Unsupported error message logged " << err.msg << std::endl;
 		}
 		
-		m_log.push_back(err);
+		m_log.push_back(std::move(err));
```

Comment:
Maybe `emplace_back()` would be a better choice here, even if it doesn't change the final assembly, since `push_back()` always calls `emplace_back()`.

Reply:
I disagree. I believe `push_back()` makes our intent of adding an already constructed object into the vector more clear, in contrast with that of `emplace_back()`, which constructs objects in-place in the vector.

Reply:
I guess you are right :/